ResourceLocator/Extension.py

```diff
@@ -0,0 +1,152 @@
+                  aws_secret_access_key='xxxx',
+                  region_name='us-east-1')
+        dynamodb = dynamodb_session.resource('dynamodb')
+        table=dynamodb.Table('GEOCODE')
+        mycity=city[0].lower()
+        mycity2=mycity
+        if (mycity.isdigit()):
+            #Pad with zeros
+            mycity2=format(mycity, '05')
+        mylog("Execute query with city "+str(mycity2))
+    
+        response = table.query(
+            KeyConditionExpression=Key('City').eq(str(mycity2)) 
+        )
+        for i in response['Items']:
+            document.add_meta_data({"mylat2":str(i['Lat'])})
+            document.add_meta_data({"mylon2":str(i['Lon'])})
+            mylog("Getting lat"+str(i['Lat']))
+            found=True
+        
+    except Exception,e:
+        mylog("Error: "+str(e))
+        
+#document.add_meta_data({"myerr":myerr})
+
+# Now lets add the random Availability
+# We could get this also from the database, but this is a demo so we create it 
+#----------------------------------------------------------
+#Initialize, 3 months back, 3 years further
+#----------------------------------------------------------
+
+if (city):
+    #Field to store the availability in
+    mylog("Start avail")
+    alldatesfield="mydateavail"
+    alldates=""
+    Date=datetime.datetime.now()
+    currenthours=random.randint(0,40)
+
+    dateBegin=Date+datetime.timedelta(days=-50)
+    dateEnd=Date+datetime.timedelta(days=3*365)
+    dateBeginP=Date+datetime.timedelta(days=10+currenthours)
+    dateEndP=Date+datetime.timedelta(days=10+currenthours+15)
+    dateBeginP2=Date+datetime.timedelta(days=60+currenthours)
+    dateEndP2=Date+datetime.timedelta(days=90+currenthours)
+    currenthours=random.randint(1,6)
+    #print dateBegin
+    #print dateEnd
+    #Put all dates by day into a string
+    tot=(dateEnd-dateBegin).days
+    
+    #print 'Total nr of days:'+str(tot)
+    for i in range( 0,tot):
+        thisdate2=dateBegin+datetime.timedelta(days=i)
+        #currenthours=random.randint(0,8)
+        hours=0
+        if (thisdate2.month % currenthours)==0:
+            hours=8
+        
+        #if ((thisdate2-dateBeginP).days>0 and (thisdate2-dateEndP).days<0):
+         #   hours=8
+        #if ((thisdate2-dateBeginP2).days>0 and (thisdate2-dateEndP2).days<0):
+        #    hours=8
+        if (hours>3):
+            alldates=alldates+";"+str(thisdate2.year)+str(thisdate2.month).zfill(2)+str(thisdate2.day).zfill(2)
+
+    #Put field in the inde        
+    mylog("End Avail")
+    document.add_meta_data({alldatesfield:alldates})
+    
+try:
+    #Add the image since this could lead to timeouts...
+    imagenr=int(document.get_meta_data_value('myrownr')[0])
+    togetnr=imagenr % 1555
+    #Get the thumbnail
+    thumbnail = document.DataStream('$thumbnail$')
+    mylog("Nr to get:"+str(togetnr)+" user:"+str(imagenr))
+    myurl="http://www.myimage.com/images/me/"+str(togetnr)+".jpg"
+    checkimg=check(myurl)   
```

Comment:
`check` _actually calls `requests.get(myurl)`, so you are calling the first image once, _without a timeout_, and then fetching it again two lines later with a timeout.

So in fact, you should remove `check` altogether.